frame/ibc-transfer/src/lib.rs

```diff
@@ -0,0 +1,576 @@
+		/// Pallet params updated
+		PalletParamsUpdated { send_enabled: bool, receive_enabled: bool },
+	}
+
+	/// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The interchain token transfer was not successfully initiated
+		TransferFailed,
+		/// Error Decoding utf8 bytes
+		Utf8Error,
+		/// Invalid asset id
+		InvalidAssetId,
+		/// Invalid Ibc denom
+		InvalidIbcDenom,
+		/// Invalid amount
+		InvalidAmount,
+		/// Invalid timestamp
+		InvalidTimestamp,
+		/// Unable to get client revision number
+		FailedToGetRevisionNumber,
+		/// Invalid params passed
+		InvalidParams,
+		/// Error opening channel
+		ChannelInitError,
+	}
+
+	#[pallet::storage]
+	#[allow(clippy::disallowed_types)]
+	/// Pallet Params used to disable sending or receipt of ibc tokens
+	pub type Params<T: Config> = StorageValue<_, PalletParams, ValueQuery>;
+
+	#[pallet::storage]
+	/// Map of asset id to ibc denom pairs (T::AssetId, Vec<u8>)
+	/// ibc denoms represented as utf8 string bytes
+	pub type IbcAssetIds<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		<T as DeFiComposableConfig>::MayBeAssetId,
+		Vec<u8>,
+		OptionQuery,
+	>;
+
+	#[pallet::storage]
+	#[allow(clippy::disallowed_types)]
+	/// ChannelIds open from this module
+	pub type ChannelIds<T: Config> = StorageValue<_, Vec<Vec<u8>>, ValueQuery>;
+
+	// Dispatchable functions allows users to interact with the pallet and invoke state changes.
+	// These functions materialize as "extrinsic", which are often compared to transactions.
+	// Dispatch able functions must be annotated with a weight and must return a DispatchResult.
+	#[pallet::call]
+	impl<T: Config> Pallet<T>
+	where
+		CurrencyId: From<<T as DeFiComposableConfig>::MayBeAssetId>,
+		AccountId32: From<T::AccountId>,
+	{
+		#[frame_support::transactional]
+		#[pallet::weight(<T as Config>::WeightInfo::transfer())]
+		pub fn transfer(
+			origin: OriginFor<T>,
+			params: TransferParams,
+			asset_id: <T as DeFiComposableConfig>::MayBeAssetId,
+			amount: <T as DeFiComposableConfig>::Balance,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+			// Check if the asset id is one of the native assets or an ibc asset id
+			// If native, get the string representation of the asset
+			let denom = if let Some(denom) = IbcAssetIds::<T>::get(&asset_id) {
+				String::from_utf8(denom).map_err(|_| Error::<T>::Utf8Error)?
+			} else {
+				match asset_id.into() {
+					CurrencyId::PICA => CurrencyId::PICA_STR.to_string(),
+					CurrencyId::KSM => CurrencyId::KSM_STR.to_string(),
+					CurrencyId::CROWD_LOAN => CurrencyId::CROWD_LOAN_STR.to_string(),
+					CurrencyId::LAYR => CurrencyId::LAYR_STR.to_string(),
+					CurrencyId::kUSD => CurrencyId::kUSD_STR.to_string(),
+					CurrencyId::USDC => CurrencyId::USDC_STR.to_string(),
+					CurrencyId::USDT => CurrencyId::USDT_STR.to_string(),
+					_ => return Err(Error::<T>::InvalidAssetId.into()),
+				}
```

Comment:
can't we add a `ToStr` trait bound on `<T as DeFiComposableConfig>::MayBeAssetId` so we avoid having to update this match statement with new assetIds?